### ISSUE TITLE
Editing existing file visibility(changes members to whom file is visible) creates additional link for the file

### ISSUE THREAD

**Issue by RishiRajSahu** [appirio-tech/connect-app#2878]
When trying to edit the file in the left bar for a Connect project by adding/removing the members to whom the file is visible, once we are done saving the changes a new link to same file with same title gets added one of them which is not working.

fyi @vikasrohit @maxceem

**Comment by vikasrohit** [comment]
@RishiRajSahu I think we need to get it fixed in upcoming release itself.

**Comment by maxceem** [comment]
@RishiRajSahu I've tried to reproduce it, but couldn't, see video https://monosnap.com/file/hqaercYdPNs6HV6hv7fFZcOmMBMaNm.
Could you share any details, maybe video?

**Comment by RishiRajSahu** [comment]
@maxceem try uploading and then editing the visibilty for members for this project https://connect.topcoder.com/projects/7306/plan of ours. I just reproduced it.

**Comment by RishiRajSahu** [comment]
@maxceem updated the title and description to be more precise.

**Comment by maxceem** [comment]
@RishiRajSahu hmm, I cannot edit file I've uploaded there https://monosnap.com/file/8xlakE6v1SIQibYS1dIiQdPYZr0mm6

Looks like there is couple of issues, and this one is server-side.

**Comment by RishiRajSahu** [comment]
@maxceem oh ! that seems a different thing, we can create a separate issue for that, you seeing this on dev as well or only on prod ?

**Comment by maxceem** [comment]
@RishiRajSahu same on DEV for `pshah_copilot`, tested with this project https://connect.topcoder-dev.com/projects/7959.

Created a new issue on project service https://github.com/topcoder-platform/tc-project-service/issues/286.

I've just tried and now get the same error for `pshah_manager`, but cannot reproduce your issue. Can reproduce your issue on DEV somehow?

**Comment by RishiRajSahu** [comment]
@maxceem yes this project https://connect.topcoder-dev.com/projects/7442

**Comment by maxceem** [comment]
Got it. This issues can only be reproduced using admin user. For all non-admin users I get an error `Only admins and the user that uploaded the docs can modify` https://github.com/topcoder-platform/tc-project-service/issues/286

**Comment by maxceem** [comment]
@RishiRajSahu I've found out another way of reproducing the same issue as you have, but it's not blocked by project service so we can fix during CF https://github.com/appirio-tech/connect-app/issues/2938

If fixing that issues wouldn't fix this one, we will come back to it, but I think this issue would be fixed as I think they have the same root.

**Comment by RishiRajSahu** [comment]
@maxceem okay we will go with your plan.

**Comment by maxceem** [comment]
@RishiRajSahu related issue https://github.com/appirio-tech/connect-app/issues/2938 is fixed in https://github.com/appirio-tech/connect-app/commit/4691058baf5de92cfafd5df638344ca1e51620b3 and I believe it fixes this issue too.

The issue was that instead of changing the selected file it sent a request for the first file. So it renamed the first file to the same name as the one which we were trying to edit. So it looked like it was duplicated.

**Comment by RishiRajSahu** [comment]
@maxceem actually the issue is duplicated, but anyway we got it fixed !

**Comment by RishiRajSahu** [comment]
Closing as fixed via https://github.com/appirio-tech/connect-app/issues/2938
